sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -112,4 +112,20 @@ class Utils {
     }
   }
 
+  /**
+   * Called when an API call fails, so that a connection error message can be shown.
+   */
+  static connectionFailed() {
```

Comment:
I'm open to suggestions. The reason I put them there was that `ApiManager` is solely responsible for talking to the backend, and shouldn't know anything about frontend notifications. Similarly, it shouldn't know about the shell element `datalab-app`.